llvm/lib/Transforms/Coroutines/CoroSplit.cpp

```diff
@@ -2207,8 +2196,53 @@ static void addPrepareFunction(const Module &M,
     Fns.push_back(PrepareFn);
 }
 
+static coro::BaseABI *CreateNewABI(Function &F, coro::Shape &S) {
+  switch (S.ABI) {
+  case coro::ABI::Switch:
+    return new coro::SwitchABI(F, S);
```

Comment:
let's try to use std::unique_ptr here

Reply:
Done. To do this I had to propagate unique_ptr to a few other lines and make use of std::move in a couple of places.